app/lib/shared/utils.dart

```diff
@@ -158,6 +173,12 @@ void validatePackageName(String name) {
   if (_reservedWords.contains(name.toLowerCase())) {
     throw new Exception('Package name must not be a reserved word in Dart.');
   }
+  final bool isLower = name == name.toLowerCase();
+  final bool matchesMixedCase =
+      knownMixedCasePackages.any((s) => s.toLowerCase() == name.toLowerCase());
```

Comment:
This is doing the linear search even if `isLower == true` (which implies it's not in the `knownMixedCasePackages` list.